python/tvm/script/parse/tir/tir.py

```diff
@@ -58,6 +60,28 @@ def visit_assign(self: Parser, node: doc.Assign) -> None:
         else:
             indices = [self.eval_expr(lhs.slice)]
         T.buffer_store(self.eval_expr(lhs.value), rhs, indices)
+    elif isinstance(lhs, doc.Name) and isinstance(rhs, PrimExpr):
+        var = T.var(rhs.dtype, lhs.id)
+        self.eval_assign(lhs, var)
+        frame = T.let(var, rhs)
+        frame.add_callback(partial(frame.__exit__, None, None, None))
+        frame.__enter__()
```

Comment:
If we want to support the case where:
```python
a, b = x + 1, y + 2
```
then things could get complicated.

in this case the best approach might be updating eval_assign's signature to:
```python
def eval_assign(
    self,
    target: doc.expr,
    source: Any,
    bind_value: Callable[[str, Any], Any], # add this as input, which generalizes `def_` here: https://github.com/junrushao1994/tvm/blob/fc062069c6a84468ae6369b52a2d1b911b1b96dd/python/tvm/script/parse/parser.py#L104
) -> None: ...
  ......
  var = bind_value(k, v)
  self.var_table.add(k, var)
  ......
```

and we supply `bind_value` using:

```python
def bind_value(name: str, value: Any) -> Any:
  if isinstance(value, PrimExpr):
    var = T.var(value.dtype)
    def_(name, var)
    frame = T.let(var, rhs)
    frame.add_callback(partial(frame.__exit__, None, None, None))
    frame.__enter__()
    return var
  elif isinstance(var, (Buffer, IterVar)):
    ...
  else:
    self.report_error("Do not know how to bind type: " + str(type(value)))
```